addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/handler/NikoHomeControlThermostatHandler.java

```diff
@@ -0,0 +1,280 @@
+                    "Niko Home Control: bridge communication not initialized when trying to execute thermostat command "
+                            + thermostatId);
+            return;
+        }
+
+        NhcThermostat nhcThermostat = nhcComm.getThermostats().get(thermostatId);
+        if (nhcThermostat == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Niko Home Control: thermostatId " + thermostatId
+                            + " does not match a thermostat in the controller");
+            return;
+        }
+
+        if (nhcComm.communicationActive()) {
+            handleCommandSelection(nhcThermostat, channelUID, command);
+        } else {
+            // We lost connection but the connection object is there, so was correctly started.
+            // Try to restart communication.
+            // This can be expensive, therefore do it in a job.
+            scheduler.submit(() -> {
+                nhcComm.restartCommunication();
+                // If still not active, take thing offline and return.
+                if (!nhcComm.communicationActive()) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Niko Home Control: communication socket error");
+                    return;
+                }
+                // Also put the bridge back online
+                nhcBridgeHandler.bridgeOnline();
+
+                // And finally handle the command
+                handleCommandSelection(nhcThermostat, channelUID, command);
+            });
+        }
+    }
+
+    private void handleCommandSelection(NhcThermostat nhcThermostat, ChannelUID channelUID, Command command) {
+        logger.debug("Niko Home Control: handle command {} for {}", command, channelUID);
+
+        if (REFRESH.equals(command)) {
+            handleStateUpdate(nhcThermostat);
+            return;
+        }
+
+        switch (channelUID.getId()) {
+            case CHANNEL_MEASURED:
+                updateStatus(ThingStatus.ONLINE);
+                break;
+
+            case CHANNEL_MODE:
+                if (command instanceof DecimalType) {
+                    // first reset the overrule
+                    // nhcThermostat.executeOverruletime(0);
+                    nhcThermostat.executeMode(((DecimalType) command).intValue());
+                }
+                updateStatus(ThingStatus.ONLINE);
+                break;
+
+            case CHANNEL_SETPOINT:
+                QuantityType<Temperature> setpoint = null;
+                if (command instanceof QuantityType) {
+                    setpoint = ((QuantityType<Temperature>) command).toUnit(CELSIUS);
+                }
+                if (setpoint == null) {
+                    try {
+                        setpoint = new QuantityType<Temperature>(new BigDecimal(command.toString()), CELSIUS);
```

Comment:
Is this still needed @wborn / @cweitkamp I have some memories that this is no longer needed, am I correct?

Reply:
Yes you are correct. With https://github.com/eclipse/smarthome/pull/5734 Eclipse SmartHome will convert any Number command without a unit to a QuantityType command with a unit. So it's no longer necessary for bindings to implement their own logic for Number commands. StringType commands should not be supported on such channels.